ruby/command-t/scanner/file_scanner.rb

```diff
@@ -15,6 +15,21 @@ class FileScanner < Scanner
       autoload :RubyFileScanner,     'command-t/scanner/file_scanner/ruby_file_scanner'
       autoload :WatchmanFileScanner, 'command-t/scanner/file_scanner/watchman_file_scanner'
 
+      def self.for_string s
+        case s
+        when 'ruby', nil # ruby is the default
+          RubyFileScanner
+        when 'find'
+          FindFileScanner
+        when 'watchman'
+          WatchmanFileScanner
+        when 'git'
+          GitFileScanner
+        else
+          raise ArgumentError, "unknown scanner type #{s.inspect}"
+        end
+      end
+
```

Comment:
I generally avoid having `new` return instances of other classes, but if you prefer that I'm not opposed.